app/service/search_index.rb

```diff
@@ -9,35 +9,57 @@ def self.log_elasticsearch_warning(message)
     end
   end
 
-  def initialize(model_name:, id:)
+  def initialize(model_name:, id:, changed: nil)
     @model_name = model_name
     @id = id
+    @changed = changed
   end
 
   def add
-    if search_index_callbacks_enabled?
-      record = model_name.constantize.find(id)
-      record.__elasticsearch__.index_document
-    else
-      log_callback_warning
+    perform { record.__elasticsearch__.index_document }
+  end
+
+  def update
+    return add unless changed.any?
+
+    perform do
+      data = elasticsearch_hash.merge(body: { doc: changes })
+      Elasticsearch::Model.client.update(data)
```

Comment:
What's the motivation to use this method of updating instead of using the method [described in the `elasticsearch-model` docs](https://github.com/elastic/elasticsearch-rails/tree/master/elasticsearch-model)?

```
record.__elasticsearch__.update_document
```